lib/pacio_test_kit/interactions_test.rb

```diff
@@ -1,6 +1,22 @@
 module PacioTestKit
   module InteractionsTest
     # TODO: All helper methods for interactions tests (CRUD) will be added here.
+
+    def update_and_validate_resource(retrieved_request, updated_status)
+      fhir_resource = retrieved_request.resource
+
+      fhir_resource_updated = fhir_resource.deep_dup
+      fhir_resource_updated.status = updated_status
+      fhir_update(fhir_resource_updated, fhir_resource.id)
+
+      assert_response_status(200)
+      assert(fhir_resource.status != resource.status,
+             'Update validation failed and resource status was not updated.')
+      assert_resource_type(resource_type)
+      assert(fhir_resource.id == resource.id, 'Resource id is not same as updated resource id.')
+      validate_response_metadata(resource, fhir_resource, 'update')
+    end
```

Comment:
Need to change the order of the checks. Regarding updating the status, I was hoping to have something like this:

```suggestion
   def update_and_validate_resource(resource_to_update)
      statuses = ['final', 'cancelled']
      new_status = statuses.find { |status| status != resource_to_update.status }
      resource_to_update.status = new_status

      fhir_update(resource_to_update, resource_to_update.id)

      perform_update_validation(resource_to_update)
    end

    def perform_update_validation(resource_to_update)
      assert_response_status(200)
      assert_resource_type(resource_type)

      msg = "Update must not change the resource ID: expected ID `#{resource_to_update.id}`, got `#{resource.id}`"
      assert resource_to_update.id == resource.id, msg

      msg = "Update failed: Expected status to be updated to `#{resource_to_update.status}`, got #{resource.status}`"
      assert resource.status == resource_to_update.status, msg

      validate_response_metadata(resource, resource_to_update, 'update')
    end
```

Note: I have not tested this.